src/ray/core_worker/core_worker.cc

```diff
@@ -1383,6 +1383,20 @@ void CoreWorker::InternalHeartbeat() {
     }
   }
 
+  // Note that Ray doesn't promise the execution order of actor retry tasks. Here,
```

Comment:
In this PR, the sequence number becomes "strictly increasing". We now always update the sequence number for resubmitted tasks. As a result, the relative order of tasks may differ between actor incarnations, because some tasks belonging to the same actor incarnation might not be included in `tasks_to_resubmit` during the same call to `InternalHeartbeat`.

Reply:
A possible follow-up: update the mechanism for selecting tasks into tasks_to_resubmit to make a better effort at preserving execution order.

Reply:
https://github.com/ray-project/ray/pull/52225